w3sper.js/tests/contract_test.js

```diff
@@ -183,3 +184,65 @@ test("contract.tx/send + events.once: transfer.deposit", async () => {
     await network.disconnect();
   }
 });
+
+test(
+  "contract.tx/send + events.once: transfer.deposit (Phoenix address)",
+  async () => {
+    const network = await Network.connect(NETWORK);
+    try {
+      // Seed & sync
+      const profiles = new ProfileGenerator(seeder);
+      const users = [await profiles.default];
+
+      const accounts = new AccountSyncer(network);
+      const addresses = new AddressSyncer(network);
+      const treasury = new Treasury(users);
+
+      // For Phoenix we need notes synced via AddressSyncer
+      await treasury.update({ addresses, accounts });
+
+      // Register driver, bind facade to BookEntry (driver is auto‑fetched)
+      network.dataDrivers.register(
+        TRANSFER_ID,
+        () => readDriverFromTarget(TRANSFER_WASM),
+      );
+      const bookentry = new Bookkeeper(treasury).as(users[0]);
+      const transfer = bookentry.contract(TRANSFER_ID, network);
+
+      // Subscribe to decoded 'deposit' before executing
+      const deposit$ = transfer.events.deposit.once();
+
+      // Execute with matching args <-> deposit, but via Phoenix (address)
+      const AMOUNT = 2n;
+      const builder = await transfer.tx.deposit(Number(AMOUNT));
+
+      const { hash } = await network.execute(
+        builder
+          .to(users[0].address)
+          .deposit(AMOUNT)
+          .gas({ limit: GAS_LIMIT }),
+      );
+      assert.ok(typeof hash === "string" && hash.length > 0);
+
+      // Check that the transaction is indeed executed
+      const executed = await network.transactions.withId(hash).once.executed();
+      const call = executed.call();
+      assert.equal(call.contract, TRANSFER_ID);
+      assert.equal(call.fn_name, "deposit");
+
+      // Get decoded event
+      const decoded = await deposit$;
+      console.log(
```

Comment:
probably a leftover?

Reply:
Yup! I copied it from the prior test, only change to this one was the addition of a Phoenix account instead of Moonlight. 

Applied all your suggestions, also in the prior test, thanks.